internal/provider/kubernetes/status_updater.go

```diff
@@ -168,40 +165,15 @@ type Updater interface {
 
 // UpdateWriter takes status updates and sends these to the UpdateHandler via a channel.
 type UpdateWriter struct {
-	log           logr.Logger
-	enabled       <-chan struct{}
 	updateChannel chan<- Update
-	// a temporary buffer to store events received before the Updater is enabled.
-	// These events will be sent to the update channel once the Updater is enabled.
-	eventsBeforeEnabled chan Update
+	wg            *sync.WaitGroup
 }
 
 // Send sends the given Update off to the update channel for writing by the UpdateHandler.
 func (u *UpdateWriter) Send(update Update) {
-	// Non-blocking receive to see if we should pass along update.
-	select {
-	case <-u.enabled:
-		u.updateChannel <- update
-	default:
-		if len(u.eventsBeforeEnabled) < cap(u.eventsBeforeEnabled) {
-			u.log.Info("received a status update while disabled, storing for later", "event", update.NamespacedName)
-			u.eventsBeforeEnabled <- update
-		} else {
-			// If the buffer is full, drop the event to avoid blocking the sender.
-			u.log.Error(errors.New("dropping status update, buffer full"), "event", update.NamespacedName)
-		}
-	}
-}
-
-// handleEventsReceivedBeforeEnabled sends the events received before the Updater was enabled to the update channel.
-func (u *UpdateWriter) handleEventsReceivedBeforeEnabled() {
-	go func() {
-		for e := range u.eventsBeforeEnabled {
-			u.log.Info("sending stored status update", "event", e.NamespacedName)
-			u.updateChannel <- e
-		}
-		close(u.eventsBeforeEnabled)
-	}()
+	// Wait until updater is ready
+	u.wg.Wait()
```

Comment:
Would the wait cause dead block like this issue?   https://github.com/envoyproxy/gateway/issues/4685

Reply:
This will only wait once during startup

Reply:
I suspect it may block because #4685 happened even when there is no wait.